sound/soc/codecs/rt715-sdca.c

```diff
@@ -88,77 +135,108 @@ static int rt715_sdca_set_amp_gain_put(struct snd_kcontrol *kcontrol,
 	struct soc_mixer_control *mc =
 		(struct soc_mixer_control *)kcontrol->private_value;
 	struct rt715_sdca_priv *rt715 = snd_soc_component_get_drvdata(component);
-	unsigned int val_l, val_r, gain_l_val, gain_r_val, loop_cnt, i, reg;
+	unsigned int gain_val, i, k_changed = 0;
 	int ret;
 
-	if (strstr(ucontrol->id.name, "FU02 Capture Volume") ||
-		strstr(ucontrol->id.name, "FU06 Capture Volume"))
-		loop_cnt = 2;
-	else if (strstr(ucontrol->id.name, "FU0E Boost") ||
-		strstr(ucontrol->id.name, "FU0C Boost"))
-		loop_cnt = 4;
-	else
-		loop_cnt = 1;
-
-	/* control value to 2s complement */
-	/* L channel */
-	gain_l_val = ucontrol->value.integer.value[0];
-	if (gain_l_val > mc->max)
-		gain_l_val = mc->max;
-	val_l = gain_l_val;
-
-	if (mc->shift == 8) {
-		gain_l_val = (gain_l_val * 10) << mc->shift;
-	} else {
-		gain_l_val =
-			((abs(gain_l_val - mc->shift) * RT715_SDCA_DB_STEP) << 8) / 1000;
-		if (val_l <= mc->shift) {
-			gain_l_val = ~gain_l_val;
-			gain_l_val += 1;
+	for (i = 0; i < 2; i++) {
+		if (ucontrol->value.integer.value[i] == rt715->kctl_2ch_orig[i])
+			continue;
+		else {
```

Comment:
the `else` is again superfluous. This could've been just
```
if (ucontrol->value.integer.value[i] != rt715->kctl_2ch_orig[i]) {
	k_changed = 1;
	break;
}
```